pkgs/by-name/lx/lxc/package.nix

```diff
@@ -51,16 +51,37 @@ stdenv.mkDerivation (finalAttrs: {
 
     # fix linking
     ./4428.diff
+
+    # Fix hardcoded path of lxc-user-nic
+    # This is needed to use unprivileged containers
+    ./user-nic.diff
   ];
 
   mesonFlags = [
-    "-Dinstall-init-files=false"
+    "-Dinstall-init-files=true"
     "-Dinstall-state-dirs=false"
     "-Dspecfile=false"
+    "-Dusernet-config-path=/etc/lxc/lxc-usernet"
+    "-Ddistrosysconfdir=${placeholder "out"}/etc/default"
+    "-Dsystemd-unitdir=${placeholder "out"}/lib/systemd/system"
     # re-enable when fixed https://github.com/lxc/lxc/issues/4427
     # "-Dtools-multicall=true"
   ];
 
+  # /run/current-system/sw/share
+  postInstall = ''
+    substituteInPlace $out/etc/default/lxc --replace "$out/etc/default" "/etc/default"
+    substituteInPlace $out/libexec/lxc/lxc-net --replace "$out/etc/default" "/etc/default"
+
+    substituteInPlace $out/share/lxc/templates/lxc-download --replace "$out/share" "/run/current-system/sw/share"
```

Comment:
Why are these being changed to current-system? It's generally better to use the full nix store path if possible, which `$out` provides. Otherwise side effects may be introduced.

Reply:
This is because, when you create a container, the container config file need to reference those config files. With future updates, the store path will change and so the old one won't work anymore, unless the user change the paths manually.

For example, the following configs are from one of my container:
```
lxc.include = /run/current-system/sw/share/lxc/config/common.conf
lxc.include = /run/current-system/sw/share/lxc/config/userns.conf
```
If instead of `/run/current-system/sw/share` we had the store path, the next time package updates the container won't be able to start. This exact case happened to me while testing the changes.